### PR TITLE
DASK GKE cuML Performance Notebook

### PR DESCRIPTION
**Pull request by drobison00**

### CODE REVIEW

**Comment by zronaghi** [comment]
I'd suggest splitting the notebook into two: 1) Setting up Dask-K8s on GKE and end-to-end workflow (NYC Taxi) with one of the cuML models 2) additional data, algorithms and benchmark scripts.

**Comment by drobison00** [comment]
@zronaghi - done. Think this addresses all the issues we've discussed. Just need to wait for 0.18 release to retest the workflows.

**Comment by drobison00** [comment]
@zronaghi Workflow is good with the latest 0.18 release. I'm able to follow the blog examples without any deviations. Its ready for public consumption and can be merged on approval.

**Comment by zronaghi** [comment]
@drobison00 could you also update the README with these notebooks (GCP and Dask)

**Review by Nanthini10 — CHANGES_REQUESTED** [review]
Looks good to me! I was able to run it fully. I had a few issues with the set-up.

I have a question - why is the `worker_counts` object a list instead of a number?

File: dask/kubernetes/specs/sched-spec.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+kind: Pod
+metadata:
+  labels:
+    dask_type: scheduler
+spec:
+  restartPolicy: Never
+  containers:
+    - image: gcr.io/nv-ai-infra/drobison/dask-unified:0.18
```
[the remark this answers is not in the document]
Comment (drobison00) [reply dask/kubernetes/specs/sched-spec.yaml:8]: Good catch. I need to update that.

File: dask/kubernetes/specs/worker-spec.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  restartPolicy: Never
+  containers:
+    - image: gcr.io/nv-ai-infra/drobison/dask-unified:0.18
+      imagePullPolicy: IfNotPresent
+      args: [ dask-cuda-worker, $(DASK_SCHEDULER_ADDRESS), --rmm-managed-memory ]
+      name: dask-cuda-worker
+      resources:
+        limits:
+          cpu: "3"
+          memory: 13G
```
[the remark this answers is not in the document]
Comment (drobison00) [reply dask/kubernetes/specs/worker-spec.yaml:15]: I'll update this, kind of surprising, the node comes with 16 GB, and usually only consumes .75 with the rest available for allocation.

**Comment by drobison00** [comment]
> Looks good to me! I was able to run it fully. There was one memory issue with the set-up YAML file.
> 
> I have a question - why is the `worker_counts` object a list instead of a number?

Worker_counts provides a list of worker counts to sweep, in this case, 8, 4, and 2 workers, which is what the blog uses for the weak scaling runs.

File: dask/kubernetes/specs/worker-spec.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Nanthini10) [reply dask/kubernetes/specs/worker-spec.yaml:15]: Yeah, that's true. I'm not sure why that happens but on the UI it says just 12.97GB is available for use.

**Review by Nanthini10 — APPROVED** [review]
Looks good to me!

**Review by zronaghi — APPROVED** [review]

**Pull request merged**